crates/red_knot_python_semantic/resources/mdtest/scopes/moduletype_attrs.md

```diff
@@ -0,0 +1,134 @@
+# Implicit globals from `types.ModuleType`
+
+## Implicit `ModuleType` globals
+
+All modules are instances of `types.ModuleType`.
+If a name can't be found in any local or global scope, we look it up
+as an attribute on `types.ModuleType` in typeshed
+before deciding that the name is unbound.
+
+```py
+reveal_type(__name__)  # revealed: str
+
+# TODO: infer types from PEP-604 union annotations
+reveal_type(__file__)  # revealed: @Todo
+reveal_type(__loader__)  # revealed: @Todo
+reveal_type(__package__)  # revealed: @Todo
+reveal_type(__spec__)  # revealed: @Todo
+
+# TODO: generics
+reveal_type(__path__)  # revealed: @Todo
+
+# TODO: this should probably be added to typeshed; not sure why it isn't?
+# error: [unresolved-reference]
+# revealed: Unbound
+reveal_type(__doc__)
```

Comment:
Probably because it's on `builtins.object`?

The awkward thing is that I think `__doc__` might be the only attribute on `object` that is accessible in every module as a global name.

I think all of them except for `__module__` are accessible on module objects as an attribute.

You'll know better than I whether it's likely we can get `__doc__` added specifically to `ModuleType` (despite it already being on `object`) to help support fixing this TODO with fewer special cases, or whether we should instead just bite the bullet and handle it as a special case.

Reply:
Yeah... I think it would be fine to add `__doc__` to `ModuleType` in typeshed if we add a comment saying why we have the "redundant" override from `object`... I'll try to put up a typeshed PR today...

Reply:
Filed https://github.com/python/typeshed/pull/12918